coord/src/rmr/rq.c

```diff
@@ -96,11 +147,14 @@ static struct queueItem *rqPop(MRWorkQueue *q) {
 void RQ_Done(MRWorkQueue *q) {
   uv_mutex_lock(&q->lock);
   --q->pending;
-  // fprintf(stderr, "Concurrent requests: %d/%d\n", q->pending, q->maxPending);
   uv_mutex_unlock(&q->lock);
 }
 
 static void rqAsyncCb(uv_async_t *async) {
+  if (!loop_th_ready) {
+    uv_async_send(async); // try again later
```

Comment:
Why we need to do it here? Isn't it makes more sense that it will be done once the connections are all connected?

Reply:
but then we lose the abstraction. The topology handling is global while this async handle belongs to a specific RQ. It is more correct from the API POV to an RQ handle to postpone itself

Reply:
But then it will enter kind of a busy loop until we will finish connecting no?

Reply:
busy wait on the event loop, yes. I don't want to implement a list of waiting RQs or break the obstruction of having only one RQ in the code. Do you have a better idea?